incubator/hnc/internal/pkg/selectors/selectors.go

```diff
@@ -33,6 +39,12 @@ func GetTreeSelector(inst *unstructured.Unstructured) (labels.Selector, error) {
 	strs := strings.Split(treeSelectorStr, ",")
 	selectorStr := ""
 	for i, str := range strs {
+		// There are cases that ok==true and treeSelectorStr is an empty string, and this
+		// will lead to selectorStr==api.LabelTreeDepthSuffix, which cannot be parsed as
+		// a valid label selector. So we need to skip the empty strings
+		if str == "" {
+			continue
+		}
```

Comment:
Putting the check here implies that any one part of the string could legitimately be empty, such as the string `,,,`, which probably shouldn't be legal.

How about putting this check on line 35/36 instead?

```
treeSelectorStr := annot[api.AnnotationTreeSelector]
if treeSelectorStr == "" {
  return nil, nil
}
```

You only need to the `x, ok := foo[bar]` form if you need to distinguish between `x` having the zero value (in the case of a string, `""`) and being missing completely. But in our case, we don't care which it is, so you can use the simpler `x := foo[bar]` form.

Reply:
done